### PR TITLE
Contact Manager as a reusable component | Features/psp 2406

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: database/mssql/scripts/dbscripts/PSP_PIMS_S21_00/Alter Down/10_DML_PIMS_STATIC_VARIABLE_VERSION.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* -----------------------------------------------------------------------------
```
Comment (devinleighsmith) [inline database/mssql/scripts/dbscripts/PSP_PIMS_S21_00/Alter Down/10_DML_PIMS_STATIC_VARIABLE_VERSION.sql:1]: @FuriousLlama why are all these IS21 dbscripts look like they were introduced by this PR?

File: frontend/src/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                ]}
+              />
+            </Col>
+            <Col>
+              <Row>
+                <Col className="pl-0">
+                  <NameInput field="summary" placeholder="Name of person or organization" />
+                </Col>
+                <Col xs="auto" className="pl-0">
+                  <CityInput field="municipality" label="City" />
```
Comment (devinleighsmith) [inline frontend/src/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent.tsx:95]: one nice thing about Styled.CityInput or StyledCityInput is that it makes it clear that this is a display only component wrapper.

File: frontend/src/components/contact/ContactManagerView/ContactManagerView.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  showSelectedRowCount?: boolean;
+  isSummary?: boolean;
+  noInitialSearch?: boolean;
+  className?: string;
+  showAddButton?: boolean;
+  showActiveSelector?: boolean;
+}
+
+/**
+ * Component that displays a list of leases within PSP as well as a filter bar to control the displayed leases.
```
Comment (devinleighsmith) [inline frontend/src/components/contact/ContactManagerView/ContactManagerView.tsx:30]: This doesn't seem right.

File: database/mssql/scripts/dbscripts/PSP_PIMS_S21_00/Alter Down/10_DML_PIMS_STATIC_VARIABLE_VERSION.sql (hunk 1 above)
Comment (FuriousLlama) [reply database/mssql/scripts/dbscripts/PSP_PIMS_S21_00/Alter Down/10_DML_PIMS_STATIC_VARIABLE_VERSION.sql:1]: that's probably git not doing the diff properly on this screen, I will force git to check again

File: frontend/src/components/contact/hooks/useAddContact.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import { logError } from 'store/slices/network/networkSlice';
+
+/**
+ * hook that adds a contact.
+ */
+const useAddContact = () => {
+  const { postPerson, postOrganization } = useApiContacts();
+  const dispatch = useDispatch();
+
+  const addPerson = async (
```
Comment (devinleighsmith) [inline frontend/src/components/contact/hooks/useAddContact.ts:17]: Maybe I'm misunderstanding, but I don't think the ability to add a contact is part of this story? This story should be creating a component that allows a contact to be added to another entity, such as a deposit.

Comment (FuriousLlama) [reply frontend/src/components/contact/hooks/useAddContact.ts:17]: that is true. I just moved all the hooks to the new location. I will remove the ones that are not used by the component directly and place them in their old place

File: frontend/src/components/contact/hooks/useUpdateContact.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import { logError } from 'store/slices/network/networkSlice';
+
+/**
+ * hook that updates a contact.
+ */
+export const useUpdateContact = () => {
+  const { putPerson } = useApiContacts();
+  const dispatch = useDispatch();
+
+  const updatePerson = async (person: IEditablePerson) => {
```
Comment (devinleighsmith) [inline frontend/src/components/contact/hooks/useUpdateContact.ts:17]: same question, this component doesn't allow existing contacts to be edited does it? It just allows them to select one.

File: backend/api/Pims.Api.csproj [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
     <UserSecretsId>0ef6255f-9ea0-49ec-8c65-c172304b4926</UserSecretsId>
-    <Version>1.1.0-21.10</Version>
+    <Version>1.1.0-21.11</Version>
```
Comment (asanchezr) [inline backend/api/Pims.Api.csproj:5]: this change should not be here... perhaps trigger a new git diff?

**Review by asanchezr — APPROVED** [review]

**Pull request merged**